### DISCUSSION TITLE
Orin NX flash issue

### DISCUSSION THREAD

**Discussion by CHNtentes** [OE4T/meta-tegra#1239] [Q&A] [answered]
Hi. I'm using Orin NX 16GB with nvme.

When I use initrd-flash to flash my image, I need to manually select "NVIDIA RCM" in UEFI boot settings, otherwise the flash process would fail.

Is this the same issue with this?
[https://github.com/OE4T/meta-tegra/wiki/initrd-flashing-support#known-issues](url) (The first one)

Thanks in advance.

**Comment by CHNtentes** [comment]
There's something wrong with my UART port, but I plug in a monitor and it displays system GUI, so Orin NX does boot into linux OS.

**Comment by madisongh** [comment]
> Is this the same issue with this?
https://github.com/OE4T/meta-tegra/wiki/initrd-flashing-support#known-issues (The first one)

That issue should be fixed since I reworked things to insert a DTB overlay (`L4TConfiguration-rcmboot.dtbo`) that sets the DefaultBootPriority UEFI variable to force the RCM boot.    Which branch are you using, and have you updated it recently?

**Reply by CHNtentes** [reply]
Thanks for your reply. I'm using kirkstone branch cloned in February whch is indeed outdated. May I ask which commit fixed this issue?

**Reply by dwalkes** [reply]
Probably one or more of these: https://github.com/OE4T/meta-tegra/search?q=RCM+Boot&type=commits

**Comment by CHNtentes** [comment]
@madisongh Hi. I updated my code and re-flashed my device. I found that there were 2 different nvme boot options in boot menu:
![image](https://user-images.githubusercontent.com/99521008/236100950-014d63d4-f963-49d9-ae40-db9aef22dc21.png)
Is this an expected behaviour? Thanks in advance.

**Reply by madisongh** [reply]
I believe so, yes.  One is for the drive, the other is for the boot partition on that drive.

**Reply by CHNtentes** [reply]
Thanks for your reply. Is it possible to remove one of those two during build time?

**Reply by CHNtentes** [reply]
NV stock L4T shows only one boot option for nvme drive.
![image](https://user-images.githubusercontent.com/99521008/236616914-084af601-28d9-41dc-8aef-e62950257235.png)

**Reply by madisongh** [reply, marked as answer]
Thanks for pointing out the discrepancy.  I tracked this down to our initrd flashing scripts not properly setting the partition type code on the `esp` partition.  Fixed with #1250.

**Reply by CHNtentes** [reply]
After updating my repo, rebuilding and reflashing, I can confirm that the issue is fixed. Thanks again for your support.
